plaidml/bridge/keras/backend_test.py

```diff
@@ -279,14 +279,15 @@ def output(self, *args):
                             atol=atol,
                             err_msg='ERR: datum={}, test={}, x=plaidml, y=theano'.format(
                                 didx, idx))
-                        for x in range(0, len(pmlr[1])):
-                            npt.assert_allclose(
-                                pmlr[1][x],
-                                thr[1][x],
-                                rtol=tol,
-                                atol=atol,
-                                err_msg='ERR: datum={}, test={}, grad, x=plaidml, y=theano'.format(
-                                    didx, idx))
+                        if do_grads:
```

Comment:
I don't think this is right, don't we want to always compare results between backends no matter if we're performing grads or not?

Reply:
I just got the code from the tensorflow case - few lines below. It is doing the same thing. The issue is that if do_grad is not true, pmlr[1] is not populated and we get a RT exception with no element at index.

Reply:
OK, I see, fair enough.